tests/conftest.py

```diff
@@ -4523,3 +4530,300 @@ def finalizer():
+            "endpoint": mcg_obj_session.s3_endpoint,
+            "ssl": ssl,
+        }
+
+    def mcg_account_factory_cleanup():
+        for account_email in created_accounts:
+            log.info(f"Deleting MCG account {account_email}")
+            mcg_obj_session.send_rpc_query(
+                "account_api", "delete_account", {"email": account_email}
+            )
+
+    request.addfinalizer(mcg_account_factory_cleanup)
+    return mcg_account_factory_implementation
+
+
+@pytest.fixture(scope="function")
+def nsfs_bucket_factory(
+    request,
+    namespace_store_factory,
+    nsfs_interface,
+    mcg_obj_session,
+    mcg_account_factory,
+    bucket_factory,
+):
+    return nsfs_bucket_factory_fixture(
+        request,
+        namespace_store_factory,
+        nsfs_interface,
+        mcg_obj_session,
+        mcg_account_factory,
+        bucket_factory,
+    )
+
+
+def nsfs_bucket_factory_fixture(
+    request,
+    namespace_store_factory,
+    nsfs_interface,
+    mcg_obj_session,
+    mcg_account_factory,
+    bucket_factory,
+):
+    created_accounts = []
+    created_rpc_buckets = []
+
+    def nsfs_bucket_factory_implementation(nsfs_obj):
+        """
+        A factory for creating an NSFS bucket and setting up all required components.
+
+        Args:
+            nsfs_obj (NSFS): An NSFS parametrization object (please see `mcg_params.py`)
+
+        """
+        # Create a PVC and namespacestore for the bucket
+        nsfs_obj.nss = namespace_store_factory(
+            nsfs_obj.method,
+            {
+                "nsfs": [
+                    (
+                        nsfs_obj.pvc_name,
+                        nsfs_obj.pvc_size,
+                        nsfs_obj.sub_path,
+                        nsfs_obj.fs_backend,
+                    )
+                ]
+            },
+        )[0]
+        # Create a deployment for mounting the PVC and accessing its filesystem
+        nsfs_deploy = nsfs_interface(nsfs_obj.nss.uls_name, nsfs_obj.mount_path)
+        deployment_app_label = nsfs_deploy.data["spec"]["selector"]["matchLabels"][
+            "app"
+        ]
+        nsfs_interface_pod = Pod(
+            **get_pods_having_label(
+                f"app={deployment_app_label}", config.ENV_DATA["cluster_namespace"]
+            )[0]
+        )
+        wait_for_pods_to_be_running(pod_names=[nsfs_interface_pod.name])
+        # Apply the necessary permissions on the filesystem
+        nsfs_interface_pod.exec_cmd_on_pod(f"chmod -R 777 {nsfs_obj.mount_path}")
+        nsfs_interface_pod.exec_cmd_on_pod(f"groupadd -g {nsfs_obj.gid} nsfs-group")
+        nsfs_interface_pod.exec_cmd_on_pod(
+            f"useradd -g {nsfs_obj.gid} -u {nsfs_obj.uid} nsfs-user"
+        )
+        nsfs_obj.interface_pod = nsfs_interface_pod
+        # Create a new MCG account
+        acc_email = (
+            create_unique_resource_name("account", "mcg").replace("-", "")
+            + "@redhat.qe"
+        )
```

Comment:
should we have "@redhat.qe" saved here? I this it should be saved in a conf or constants file

Reply:
Email is not supported by the CRD, so it was removed entirely